src/Chapter4.hs

```diff
@@ -488,10 +494,29 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+instance Applicative List where
+   pure :: a -> List a
+   pure a = Cons a Empty
 
+   (<*>) :: List (a -> b) -> List a -> List b
+   fs <*> xs =  flatten (fmap (\x -> fmap (\f -> f x) fs) xs)
+   _ <*> _ = Empty
 
-{- |
-=🛡= Monad
+flatten :: List (List b) -> List b
+flatten (Cons xs Empty) = xs
+flatten (Cons Empty (Cons xs xxs)) = xs +.+ flatten xxs
+flatten (Cons xs@(Cons _ _) yys@(Cons _ _)) = xs +.+ flatten yys
+flatten _ = Empty
```

Comment:
The implementation of this function can be simplified. You've correctly implemented a function to append two lists (`+.+` operator in your case). So the `flatten` function can be implemented simpler using just two patterns:

```haskell
flatten :: List (List b) -> List b
flatten Empty = ...
flatten (Cons x xs) = ...
```